tools/castle-editor/code/castleundosystem.pas

```diff
@@ -11,6 +11,20 @@ interface
   Classes, Generics.Collections,
   CastleUtils;
 
+type
+  TUndoCommentPriority = (
+    { Low priority comments will get overwritten by a higher priority comments
+      which correspond to a more specific description of what has happened,
+      but are not always available. }
+    ucLow = 0,
+    { High priority comments usually describe what exactly happened
+      and should overwrite less specific comments. }
+    ucHigh = 1,
+    { Highest priority comments should never be overwritten.}
+    ucHighest = 31
+    );
```

Comment:
Do not use explicit numeric values for the enums here. I see no reason for it. And you should not need `ucHighest`, as you can use `High(TUndoCommentPriority)`.

So this definition can be simpler, just 

```pascal
  TUndoCommentPriority = (
    { Low priority comments will get overwritten by a higher priority comments
      which correspond to a more specific description of what has happened,
      but are not always available. }
    ucLow,
    { High priority comments usually describe what exactly happened
      and should overwrite less specific comments. }
    ucHigh
);
```